### PR TITLE
Add test fixture for k8s docker cluster

### PR DESCRIPTION
**Pull request by carolineechen**

### CODE REVIEW

**Comment by carolineechen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1185?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1185** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1185?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1180** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1180?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1174** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1174?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1173** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1173?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1172** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1172?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1171** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1171?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1170** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1170?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @carolineechen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tests/fixtures/on_demand_cluster_fixtures.py [hunk 1]
```diff
@@ -119,6 +120,24 @@ def ondemand_k8s_cluster(request):
...
+
+    if not kube_config_path.exists():
+        pytest.skip("no kubeconfig found")
+
+    args = {
+        "name": "k8s-docker-cpu",
+        "provider": "kubernetes",
+        "instance_type": "CPU:1",
+        "memory": ".2",
+        "image_id": "docker:ubuntu:18.04",
```
Comment (dongreenberg) [inline tests/fixtures/on_demand_cluster_fixtures.py:135]: Maybe we should do a more specific one (e.g. a Ray or SkyPilot one) to make the testing have to accomodate existing Python / package versions already installed?

File: tests/test_resources/test_clusters/test_on_demand_cluster.py [hunk 2]
```diff
@@ -115,6 +117,7 @@ def test_restart_does_not_change_config_yaml(self, cluster):
             assert config_yaml_content_after_restart == config_yaml_content
 
     @pytest.mark.level("minimal")
+    @pytest.mark.skip("invalid")
```
Comment (dongreenberg) [inline tests/test_resources/test_clusters/test_on_demand_cluster.py:120]: Is this test broken?

**Review by dongreenberg — APPROVED** [review]

File: tests/test_resources/test_clusters/test_on_demand_cluster.py (hunk 2 above)
Comment (carolineechen) [reply tests/test_resources/test_clusters/test_on_demand_cluster.py:120]: oops did not mean to submit this to the PR, but k8s were hanging on this at one point. it is failing on main release tests with ` ModuleNotFoundError: No module named 'tests.conftest'`

Comment (carolineechen) [reply tests/test_resources/test_clusters/test_on_demand_cluster.py:120]: oops did not mean to submit this to the PR, but k8s tests were hanging on this at one point so I skipped it to let the tests finish. but does look like it is failing on main release tests with ` ModuleNotFoundError: No module named 'tests.conftest'`

**Comment by carolineechen** [comment]
### Merge activity

* **Aug 29, 5:33 PM EDT**: @carolineechen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/1185).

**Pull request merged**